File: testdata/jennies/rawtypes/enums_as_map_index/JavaRawTypes/enums_as_map_index/SomeStruct.java
package enums_as_map_index;

import java.util.Objects;
import java.util.HashMap;
import java.util.Map;

public class SomeStruct {
    public Map<StringEnum, String> data;
    public SomeStruct() {
        this.data = new HashMap<>();
    }
    public SomeStruct(Map<StringEnum, String> data) {
        this.data = data;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof SomeStruct)) return false;
        SomeStruct o = (SomeStruct) other;
        if (!Objects.equals(this.data, o.data)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.data);
    }
}

File: testdata/jennies/rawtypes/enums_as_map_index/PHPRawTypes/src/EnumsAsMapIndex/SomeStruct.php
<?php

namespace Grafana\Foundation\EnumsAsMapIndex;

class SomeStruct implements \JsonSerializable
{
    /**
     * @var array<\Grafana\Foundation\EnumsAsMapIndex\StringEnum, string>
     */
    public array $data;

    /**
     * @param array<\Grafana\Foundation\EnumsAsMapIndex\StringEnum, string>|null $data
     */
    public function __construct(?array $data = null)
    {
        $this->data = $data ?: [];
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->data != $other->data) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{data?: array<string, string>} $inputData */
        $data = $inputData;
        return new self(
            data: $data["data"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->data = $this->data;
        return $data;
    }
}
